src/daft-io/src/object_store_glob.rs

```diff
@@ -309,6 +312,14 @@ fn _should_return(fm: &FileMetadata) -> bool {
         {
             false
         }
+        // Do not return Spark marker files
+        FileType::File
+            if MARKER_SUFFIXES
+                .iter()
+                .any(|suffix| fm.filepath.to_lowercase().ends_with(suffix)) =>
```

Comment:
for `_metadata`, `_common_metadata`, and `_success` this may be bug prone. 
Imagine you have a parquet file called `image_metadata`. this would filter that file out. Instead you want to find an exact match of `filename` where `filename` is `basepath / filename` for those 3.